internal/backend/runtime/omni/controllers/omni/internal/task/machine/poll.go

```diff
@@ -340,3 +346,48 @@ func pollExtensions(ctx context.Context, c *client.Client, info *Info) error {
 
 	return nil
 }
+
+func pollMaintenanceConfig(ctx context.Context, c *client.Client, info *Info) error {
+	if !info.MaintenanceMode {
+		return nil
+	}
+
+	conf, err := safe.StateGetByID[*config.MachineConfig](ctx, c.COSI, config.V1Alpha1ID)
+	if err != nil {
+		// if we successfully probed the config and found out that it is not set, we mark it as ready by setting it to a non-nil value by empty config.
```

Comment:
comment doesn't match as this is check on non-nil error

Reply:
Actually it is intended, but I changed the wording. The idea here is: if the error is a NotFound, it is not an unexpected error, our call actually succeeded in a way - we just found out that there is no config set at all. In this case, we know what the maintenance config is - it is explicitly empty. So we set the value to that.

Reply:
i got the intention, just that the comment was confusing since it said if it was successful